assembl/static2/js/app/components/debate/tableOfIdeas/ideas.jsx

```diff
@@ -0,0 +1,119 @@
+    this.goBackToParents = this.goBackToParents.bind(this);
+  }
+
+  setSelectedIdeas(selectedIdeaId, ideaLevel, ideaIndex) {
+    const nbLevel = this.state.selectedIdeasId.length;
+    const ideasArray = this.state.selectedIdeasId;
+    this.setState({ selectedIdeaIndex: ideaIndex, ideaLevel: ideaLevel });
+    if (ideasArray.indexOf(selectedIdeaId) <= -1 && ideaLevel === nbLevel) {
+      ideasArray.push(selectedIdeaId);
+      this.setState({
+        selectedIdeasId: ideasArray
+      });
+    } else if (ideaLevel < nbLevel) {
+      const nbToRemove = this.state.selectedIdeasId.length - ideaLevel;
+      ideasArray.splice(ideaLevel, nbToRemove, selectedIdeaId);
+      this.setState({
+        selectedIdeasId: ideasArray
+      });
+    }
+  }
+
+  getIdeaChildren(selectedIdeaId) {
+    const { ideas } = this.props;
+    return ideas.filter(idea => idea.parentId === selectedIdeaId);
+  }
+
+  getIdeaParents(ideaLevel) {
+    const { ideas } = this.props;
+    const { selectedIdeasId } = this.state;
+    const ancestor = selectedIdeasId[ideaLevel];
+    return ideas.filter(idea => ancestor === idea.ancestors[0]);
+  }
+
+  goBackToParents() {
+    const { selectedIdeasId, ideaLevel } = this.state;
+    const ideasArray = selectedIdeasId;
+    ideasArray.pop();
+    this.setState({
+      selectedIdeasId: ideasArray,
+      goBack: true,
+      ideaLevel: ideaLevel - 1,
+      selectedIdeaIndex: 0
+    });
+  }
+
+  render() {
+    const isMobile = window.innerWidth < SMALL_SCREEN_WIDTH;
+    const { identifier } = this.props;
+    return (
+      <section className={classnames('themes-section', 'ideas-section', { 'mobile-ideas-section': isMobile })}>
+        <Grid fluid className={classnames('background-grey', { 'no-padding': isMobile })}>
+          <div className="max-container">
+            {!isMobile && <PageTitle />}
+            {isMobile && this.state.ideaLevel === 0 && <PageTitle />}
+            {isMobile &&
+              this.state.ideaLevel > 0 && (
+                <div className="ideas-back-arrow" onClick={this.goBackToParents}>
+                  <span className="assembl-icon-down-small color" />
+                </div>
+              )}
+            <div className={classnames('content-section', { 'mobile-content-section': isMobile })}>
+              {!isMobile
+                ? this.state.selectedIdeasId.map((ideaId, index) => (
+                  <IdeasLevel
+                    key={`ideas-level-${index}`}
+                    ideas={this.getIdeaChildren(ideaId)}
+                    identifier={identifier}
+                    setSelectedIdeas={this.setSelectedIdeas}
+                    nbLevel={this.state.selectedIdeasId.length}
+                    ideaLevel={index + 1}
+                    selectedIdeasId={this.state.selectedIdeasId}
+                    selectedIdeaIndex={this.state.selectedIdeaIndex}
+                  />
+                ))
+                : this.state.selectedIdeasId.map((ideaId, index) => (
```

Comment:
You can destructure ideaLevel and selectedIdeasId from state to lines a bit shorter. obviously no big deal

Reply:
yep ;)